pkg/schema/check-strategy.go

```diff
@@ -31,55 +32,48 @@ var ErrInvalidSchemaVersion = errors.New("invalid schema version")
 // ValidateSchemaVersion checks the specified schema version against the supported version,
 // returning if the result is a warning only. Warnings are returned when the versions are not an exact match.
 // A warning is not returned when CheckStrategyNone is used.
-func ValidateSchemaVersion(strategy CheckStrategy, supported string, specified string) (bool, error) {
+func ValidateSchemaVersion(strategy CheckStrategy, supported *semver.Constraints, specified string, defaultVersion *semver.Version) (bool, error) {
 	if specified == "" {
 		specified = "(none)"
 	}
 	baseMessage := fmt.Errorf("the schema version is %s but the supported schema version is %s. See https://getporter.org/reference/file-formats/#supported-versions for more details: %w",
 		specified, supported, ErrInvalidSchemaVersion)
 
+	specifiedV, err := semver.NewVersion(specified)
+	if err != nil {
+		isWarning := strategy == CheckStrategyNone
+		return isWarning, fmt.Errorf("%s is not a valid semantic version: %w", specified, ErrInvalidSchemaVersion)
+	}
+
+	isMatch := supported.Check(specifiedV)
```

Comment:
Since I have changed it to use a constraint, how about I update it to indicate that the constrain is satisfied? e.g. `isSchemaVersionConstraintSatisfied` (long but maybe helpful)